app/src/main/java/ch/epfl/sweng/SDP/utils/ImageStorageManager.java

```diff
@@ -0,0 +1,75 @@
+package ch.epfl.sweng.SDP.utils;
+
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.media.MediaScannerConnection;
+import android.os.Build;
+import android.os.Environment;
+import android.support.annotation.RequiresApi;
+import android.util.Log;
+import android.widget.Toast;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import ch.epfl.sweng.SDP.Activity;
+
+/**
+ * This class is responsible for saving images to the device hard drive.
+ */
+public class ImageStorageManager {
+
+    ImageStorageManager() {
+        // Empty constructor
+    }
+
+    /**
+     * Saves an image to the device file system.
+     * @param image the image to save.
+     * @param image_name the filename of the image.
+     * @param activity the activity.
+     * @param context the context.
+     */
+    @RequiresApi(api = Build.VERSION_CODES.O)
+    public static void saveImage(Bitmap image, String image_name, final Activity activity, final Context context) {
+
+        // Get image dirctory
+        String root = Environment.getExternalStoragePublicDirectory(
+                Environment.DIRECTORY_DCIM).toString()+ "/Camera/Gyrodraw/";
+        File myDir = new File(root);
+        myDir.mkdirs();
+        String fname = "Image-" + image_name + ".png";
+        File file = new File(myDir, fname);
+
+        Log.d("ImageStorageManager",  "Saving image: " + file.getPath());
+
+        if (file.exists()) {
+            file.delete();
+        }
+
+        // Save image in file directory
```

Comment:
Instead of explaining what the next chunk of code does, you could maybe put this into an own method?

Reply:
Yeah you are right. I will extract code to a new method to make it more modular.